src/main/kotlin/no/nav/k9punsj/journalpost/JournalpostkopieringService.kt

```diff
@@ -0,0 +1,144 @@
+            FordelPunsjEventDto(
+                aktørId = kopierJournalpostDto.til,
+                journalpostId = nyJournalpostId.toString(),
+                type = K9FordelType.KOPI.kode,
+                ytelse = k9FagsakYtelseType.kode,
+                gosysoppgaveId = null
+            )
+        )
+
+        return nyJournalpostId
+    }
+
+    private suspend fun validerJournalpostKopiering(
+        journalpostId: JournalpostId,
+        kopierJournalpostDto: KopierJournalpostDto,
+    ): Pair<SafDtos.Journalpost, FagsakYtelseType> {
+
+        val journalpost = journalpostRepository.hentHvis(journalpostId.toString())
+            ?: throw KanIkkeKopieresErrorResponse("Finner ikke journalpost i DB.")
+
+        val safJournalpost = safGateway.hentJournalpostInfo(journalpostId.toString())
+            ?: throw KanIkkeKopieresErrorResponse("Finner ikke SAF journalpost.")
+
+        if (safJournalpost.erUtgående) {
+            throw KanIkkeKopieresErrorResponse("Ikke støttet journalposttype: ${safJournalpost.journalposttype}")
+        }
+
+        if (!safJournalpost.kanKopieres) {
+            throw KanIkkeKopieresErrorResponse("Kan ikke kopieres. $journalpost.")
+        }
+
+        if (journalpost.type != null && journalpost.type == K9FordelType.INNTEKTSMELDING_UTGÅTT.kode) {
+            throw KanIkkeKopieresErrorResponse("Kan ikke kopier journalpost med type inntektsmelding utgått.")
+        }
+
+        val k9FagsakYtelseType: FagsakYtelseType =
+            kopierJournalpostDto.ytelse?.somK9FagsakYtelseType() ?: journalpost.ytelse?.let {
+                val punsjFagsakYtelseType = PunsjFagsakYtelseType.fromKode(it)
+                journalpost.utledK9sakFagsakYtelseType(punsjFagsakYtelseType.somK9FagsakYtelseType())
+            } ?: throw KanIkkeKopieresErrorResponse("Mangler ytelse for journalpost.")
+
+        val støttedeYtelseTyperForKopiering = listOf(
+            FagsakYtelseType.OMSORGSPENGER_KS,
+            FagsakYtelseType.PLEIEPENGER_SYKT_BARN,
+            FagsakYtelseType.PLEIEPENGER_NÆRSTÅENDE
+        )
+
+        if (!støttedeYtelseTyperForKopiering.contains(k9FagsakYtelseType)) {
+            throw KanIkkeKopieresErrorResponse("Støtter ikke kopiering av ${k9FagsakYtelseType.navn} for relaterte journalposter")
+        }
+
+        return Pair(safJournalpost, k9FagsakYtelseType)
+    }
+
+    private suspend fun hentEllerOpprettSaksnummer(
+        journalpostId: String,
+        kopierJournalpostDto: KopierJournalpostDto,
+        k9SakGrunnlag: HentK9SaksnummerGrunnlag,
+    ): String {
+        // Henter eller oppretter saksnummer for den originale personen
+        val fraSaksnummer = k9SakService.hentEllerOpprettSaksnummer(k9SakGrunnlag)
+
+        // Sjekker om journalposten kopieres til samme person og logger hvis så er tilfelle
+        if (kopierJournalpostDto.fra == kopierJournalpostDto.til) {
+            logger.info("Kopierer journalpost: $journalpostId til samme person.")
```

Comment:
Eg kjenner ikkje så godt til caset her, men det er jo mulig å ha to saker for samme søker.

Reply:
Det er flere caser her.

- Kopier JP 1 for søker 1, barn 1 --> JP 2: søker 1, barn 1 --> sak 1
- Kopier JP 1 for søker 1, barn 2 --> JP2: søker 1, barn 2 --> sak 2
- Kopier JP 1 for søker 1, annen part, barn 1 --> JP 2: annen part, barn 1 --> sak 2

Journalposten blir i prinsipp enten kopiert på samme person eller annen person.

Reply:
Kan fra og til her vere både søker og barn?

Reply:
Nei, kun fra søkere. Det er det annenPart og barn feltene brukes til.

`Fra` og `Til` brukes for journalpostene. `Barn` sammen med søker og/eller `annenPart` brukes for å opprette/hente sak.